tests/models/test_sasrec.py

```diff
@@ -582,3 +593,349 @@ def test_get_dataloader_recommend(
+                [15, "f2", "f2val2"],
+                [17, "f1", "f1val2"],
+                [17, "f2", "f2val3"],
+                [16, "f1", "f1val2"],
+                [16, "f2", "f2val3"],
+                [11, "f3", 0],
+                [12, "f3", 1],
+                [13, "f3", 2],
+                [14, "f3", 3],
+                [15, "f3", 4],
+                [17, "f3", 5],
+                [16, "f3", 6],
+            ],
+            columns=["id", "feature", "value"],
+        )
+        ds = Dataset.construct(
+            INTERACTIONS,
+            item_features_df=item_features,
+            cat_item_features=["f1", "f2"],
+        )
+        return ds
+
+    def test_device(self, dataset_item_features: Dataset) -> None:
+        cat_item_embeddings = CatFeaturesItemNet.from_dataset(dataset_item_features, n_factors=5, dropout_rate=0.5)
+        assert cat_item_embeddings.device == torch.device("cpu")
+
+    def test_feature_catalogue(self, dataset_item_features: Dataset) -> None:
+        cat_item_embeddings = CatFeaturesItemNet.from_dataset(dataset_item_features, n_factors=5, dropout_rate=0.5)
+        expected_feature_catalogue = torch.arange(
+            0, cat_item_embeddings.n_cat_features, device=cat_item_embeddings.device
+        )
+        assert torch.equal(cat_item_embeddings.feature_catalogue, expected_feature_catalogue)
+
+    def test_get_dense_item_features(self, dataset_item_features: Dataset) -> None:
+        items = torch.from_numpy(
+            dataset_item_features.item_id_map.convert_to_internal(INTERACTIONS[Columns.Item].unique())
+        )
+        cat_item_embeddings = CatFeaturesItemNet.from_dataset(dataset_item_features, n_factors=5, dropout_rate=0.5)
+
+        actual_feature_dense = cat_item_embeddings.get_dense_item_features(items)
+        expected_feature_dense = torch.tensor(
+            [
+                [1.0, 0.0, 1.0, 0.0, 0.0],
+                [1.0, 0.0, 0.0, 1.0, 0.0],
+                [0.0, 1.0, 1.0, 0.0, 0.0],
+                [1.0, 0.0, 0.0, 0.0, 1.0],
+                [0.0, 1.0, 0.0, 1.0, 0.0],
+                [0.0, 1.0, 0.0, 0.0, 1.0],
+            ],
+            device=cat_item_embeddings.device,
+        )
+
+        assert torch.equal(actual_feature_dense, expected_feature_dense)
+
+    @pytest.mark.parametrize("n_factors", (10, 100))
+    def test_create_from_dataset(self, n_factors: int, dataset_item_features: Dataset) -> None:
+        cat_item_embeddings = CatFeaturesItemNet.from_dataset(
+            dataset_item_features, n_factors=n_factors, dropout_rate=0.5
+        )
+
+        actual_item_features = cat_item_embeddings.item_features
+        actual_n_items = cat_item_embeddings.n_items
+        actual_n_cat_features = cat_item_embeddings.n_cat_features
+        actual_embedding_dim = cat_item_embeddings.category_embeddings.embedding_dim
+
+        expected_item_features = dataset_item_features.item_features
+        # TODO: remove after adding Dense Features support
```

Comment:
Dense features do not support categories. So there will be no support for CatFeaturesItemNet if features are dense.
@feldlime please correct me if I'm wrong